### PR TITLE
Pensions form bug fixes and Cypress test

### PR DESCRIPTION
**Pull request by U-DON**
## Description
In the process of migrating Nightwatch tests to GitHub Actions, the test for the pensions form was giving me issues. Rather than debug it in Nightwatch, I decided to just convert the test to Cypress using the form tester.

Thanks to the automatic accordion expansion and aXe checks included with the form tester, I actually uncovered some bugs with the app's review page. These are also fixed by the changes.
1. The app would throw an exception when expanding a particular chapter's accordion.
2. The markup in the expanded accordions yielded accessibility errors.
    - The issues were primarily about the use of `<dl>` and its structure.
    - Mostly due to multiple levels `<div>` wrapping the `<dd>` and `<dt>`.
    - [Slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1623081904050000)

## Testing done
Ran the test locally. The test should pass in CI.

## Acceptance criteria
- [ ] Pension form's review page should pass aXe check.
- [ ] Pension form's review page accordions should be able to expand without throwing exceptions.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/pensions/config/form.js
Comment (U-DON) [inline src/applications/pensions/config/form.js:709]: I don't think this is a pattern supported by the Forms Library. This is probably just effectively passing 0 in both cases, but the title should be updated by the `uiSchema` configuration anyway, so this might not matter.

File: src/applications/pensions/config/form.js
Comment (U-DON) [inline src/applications/pensions/config/form.js:1455]: This hacky pattern seems to be used in some other forms (found by searching `customTitle`).

[The logic for this can be found in this code](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/review/ObjectField.jsx#L125).

**Comment by alexandec** [comment]
It seems strange that frontend-review-group is the only codeowner here. Wondering whether codeowners needs to be updated for these files, and if we can try to get a review from the team that works on this.

**Comment by U-DON** [comment]
@Mottie: Does pensions belong to your team? If so, do you mind reviewing it? Or if not, feel free to correct me or pass this along to the right folks. 😅

**Review by 1Copenut — APPROVED** [review]
LGTM! I ran the test locally and mocked my own pension to test with axe browser plugin manually. Zero axe violations.

**Comment by Mottie** [comment]
I'm pretty sure we don't own pensions - @vhenry07?

**Comment by alexandec** [comment]
FYI looks like Cypress tests are stuck on `21-527EZ pension benefits form -> maximal-test -> fills the form`

**Comment by vhenry07** [comment]
> I'm pretty sure we don't own pensions - @vhenry07?

Correct. The Claims & Appeals team does not own Pensions.

File: src/applications/pensions/tests/00-all-fields.e2e.spec.js
Comment (U-DON) [inline src/applications/pensions/tests/00-all-fields.e2e.spec.js:25]: It seems that this test originally intended for the dependents' ages to be within or outside of certain ranges for functions such as `isBetween18And23` and `isEligibleForDisabilitySupport`.

The output of those functions changed while the birth dates from the test data remained static. Eventually, those outputs met certain conditions, requiring some fields to be filled that weren't accounted for in the original test data. So this fix to dynamically update the test data was necessary to pass the test.

It seems that at this point in time, the ages are completely out of the ranges of those conditions, so the new Cypress test won't fail in the same way anymore. I will not be updating the ages to match any exact criteria and will leave it for future owners to amend the test data as needed.

**Review by alexandec — APPROVED** [review]

**Pull request merged**
